pop-api/src/v0/mod.rs

```diff
@@ -1,21 +1,37 @@
 use crate::{
-	primitives::storage_keys::{ParachainSystemKeys, RuntimeStateKeys},
-	BlockNumber, PopApiError,
+	build_extension_method,
+	constants::{DISPATCH, READ_STATE},
+	primitives::Error,
+	StatusCode,
 };
+use ink::env::chain_extension::ChainExtensionMethod;
 
-pub mod balances;
-pub mod cross_chain;
-pub mod nfts;
-pub mod state;
+/// APIs for fungible tokens.
+#[cfg(feature = "fungibles")]
+pub mod fungibles;
 
-pub fn relay_chain_block_number() -> Result<BlockNumber, PopApiError> {
-	state::read(RuntimeStateKeys::ParachainSystem(ParachainSystemKeys::LastRelayChainBlockNumber))
+pub(crate) const V0: u8 = 0;
+
+impl From<StatusCode> for Error {
+	fn from(value: StatusCode) -> Self {
+		value.0.into()
+	}
+}
+
+// Helper method to build a dispatch call.
+//
+// Parameters:
+// - 'module': The index of the runtime module.
+// - 'dispatchable': The index of the module dispatchable functions.
+fn build_dispatch(module: u8, dispatchable: u8) -> ChainExtensionMethod<(), (), (), false> {
```

Comment:
Maybe we just use a type alias for the ChainExtensionMethod across the crate to just simplify as `ChainExtensionMethod`?